Camera.py

```diff
@@ -0,0 +1,428 @@
+#***************************************************************************
+#*                                                                         *
+#*   Copyright (c) 2020 Howetuft <howetuft@gmail.com>                      *
+#*                                                                         *
+#*   This program is free software; you can redistribute it and/or modify  *
+#*   it under the terms of the GNU Lesser General Public License (LGPL)    *
+#*   as published by the Free Software Foundation; either version 2 of     *
+#*   the License, or (at your option) any later version.                   *
+#*   for detail see the LICENCE text file.                                 *
+#*                                                                         *
+#*   This program is distributed in the hope that it will be useful,       *
+#*   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+#*   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+#*   GNU Library General Public License for more details.                  *
+#*                                                                         *
+#*   You should have received a copy of the GNU Library General Public     *
+#*   License along with this program; if not, write to the Free Software   *
+#*   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+#*   USA                                                                   *
+#*                                                                         *
+#***************************************************************************
+
+"""This module implements a Camera object, which allows to take a snapshot
+of Coin Camera settings and use them later for rendering"""
+
+from collections import namedtuple
+from math import degrees, radians
+import shlex
+
+from PySide2.QtWidgets import QAction
+from PySide2.QtCore import QT_TRANSLATE_NOOP, QObject, SIGNAL
+from pivy import coin
+
+import FreeCAD
+import FreeCADGui
+import Part
+
+# ===========================================================================
+
+def createCamera():
```

Comment:
I'm not saying that it is wrong how you write it here, but maybe you could use PEP8 guidelines for Python. Meaning that functions should be in `snake_case()`, rather than `PascalCase()`. In other Python workbenches we try to slowly migrate to PEP8, even if the current code is not in this style. Obviously those functions defined in C++ cannot be renamed, but user defined functions should probably use PEP8 style.

Reply:
Hello vox-fc,

Thank you for your comments! I'm a big fan of PEP 8/ PEP 20, so I should be very happy to rename my functions into snake_case style (except framework callbacks, for the reasons you mention...). That was my original intention, but I had fallen back to CamelCase for consistency with the rest of the workbench.
I shall soon post a commit for that.

By the way, I have 2 questions about that:
- PEP8 requires module names to be in lowercase: should I rename my module 'camera.py' instead of 'Camera.py'?
- Would it be desirable and above all acceptable (not offendable for original writers, not harmful for current users) that I refactor the whole workbench to make it PEP8 compliant?

Reply:
This is a hard question and not always possible, since some (many) third-party libs that we use in FreeCAD (ex. PySide or pivy) do use CamelCase. This is common with projects that mix python and C++. So inevitably the FreeCAD code will still hold CamelCase stuff all around. So this is not a golden rule. But as @vocx-fc says we should try to use snake case as much as possible, were possible. I'm also all for code refactoring!

Reply:
> * PEP8 requires module names to be in lowercase: should I rename my module 'camera.py' instead of 'Camera.py'?

Is `Camera.py` a new module that previously didn't exist? Then yes, absolutely you should use lowercase. If you are modifying a previously existing module, like `Render.py`, then you have to make a conscious decision on whether you are breaking past code or not. Maybe this module is used internally and not directly by users, so renaming it does not really affect users, but if it does, you have to think whether it is worth it or not.

> * Would it be desirable and above all acceptable (not offendable for original writers, not harmful for current users) that I refactor the whole workbench to make it PEP8 compliant?

"Offensive to original authors" doesn't really matter. You are submitting a pull request. If the administrator likes your changes, he will accept them, regardless of what the previous author wrote. That's just how free software works. We constantly overwrite what others wrote with the intention of improving it.

"Harmful to current users", well, this is a bigger issue. My personal belief is that we should be a bit cautious with breaking backwards compatibility. We could re-write completely many things, but that will probably break older code and piss off our users, so in this case, we need to carefully make changes to maintain old things working.

For example, if there is an internal variable `someRenderVariable` that is not used outside a function, then this is not publicly accessible to users, so we can rename it to `some_render_variable` without issues.

The problem is if we have a function like `makeCamera` that is intended to be used by users. This function will probably be already in scripts, so if we change it to `make_camera` it will  break those scripts. My suggestion would be to rename the function to `make_camera` but still provide an alias referencing the old function.
```py
def make_camera():
...

makeCamera = make_camera
```
This is essentially what I did with [#2829](https://github.com/FreeCAD/FreeCAD/pull/2829) and [#2830](https://github.com/FreeCAD/FreeCAD/pull/2830), when I moved many functions from `Draft.py` to their own modules. The functions have a new name, but we provide the old names for compatibility. In the future we would like to deprecate the old functions, but that will take some time, years even.

Reply:
@vocx-fc:
Yes, "camera.py" is a new module, so I'm renaming it "camera.py" right now.
I've taken note of your suggestion for refactoring (keeping aliases on the old names). I think the Render module is not used much as an API, it should mostly be accessed by GUI (toolbar), but I shall be careful about it.